test/remove.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var S = require('..');
+
+var assert = require('assert');
+var eq = require('./internal/eq');
+
+
+test('remove', function() {
+
+  eq(typeof S.remove, 'function');
+  eq(S.remove.length, 2);
+  eq(S.remove.toString(), 'remove :: String -> StrMap a -> StrMap a');
+
+  function testOriginalReturned(key, strMap) {
+    var strMapAfterDeletion = S.remove(key, strMap);
+    eq(strMap, strMapAfterDeletion);
+    assert.strictEqual(strMap, strMapAfterDeletion);
+  }
```

Comment:
Even if we decide to return the input in some cases, I don't think this should be enforced by the test suite. If, on the other hand, we decide it's important to return a new string map in all cases we could consider asserting that the output is *not* identical to the input.

Reply:
If we're specifying that `remove` returns the original object, shouldn't we test for it? We could just not specify whether removing a non-existing key will return the original object. Or we could _always_ return a new string map.